samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -274,22 +434,22 @@ static void base_recv_cb(struct bt_bap_broadcast_sink *sink, const struct bt_bap
 			printk("\tIndex 0x%02x\n", index);
 
 			base_bis_index_bitfield |= BIT(index);
-		}
 #if defined(CONFIG_LIBLC3)
-		int ret;
-		const struct bt_audio_codec_cfg *codec_cfg = &base->subgroups[i].codec_cfg;
+			int ret;
+			const struct bt_audio_codec_cfg *codec_cfg = &base->subgroups[i].codec_cfg;
 
-		if (codec_cfg->id != BT_HCI_CODING_FORMAT_LC3) {
-			printk("unsupported codec 0x%02x", codec_cfg->id);
-			return;
-		}
+			if (codec_cfg->id != BT_HCI_CODING_FORMAT_LC3) {
+				printk("Unsupported codec 0x%02x", codec_cfg->id);
+				return;
+			}
 
-		ret = lc3_enable(codec_cfg);
-		if (ret < 0) {
-			printk("Error: cannot enable LC3 codec: %d", ret);
-			return;
-		}
+			ret = lc3_enable(codec_cfg, &streams[j]);
+			if (ret < 0) {
+				printk("Error: cannot enable LC3 codec: %d", ret);
+				return;
+			}
```

Comment:
This won't work if `base->subgroup_count > 1` as you will be initializing `streams[0]` more than once